### PR TITLE
[SPARK-43219][SQL][DOCS] Add `INSERT INTO REPLACE WHERE` statement into website

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
`INSERT INTO REPLACE` statement be supported in #38404 , but can't be found in website

### Why are the changes needed?
Add doc for feature we supported.

### Does this PR introduce _any_ user-facing change?
image
image

### How was this patch tested?
Unnecessary

### CODE REVIEW

**Comment by Hisoka-X** [comment]
cc @cloud-fan @HyukjinKwon @MaxGekk

**Review by MaxGekk — COMMENTED** [review]
@Hisoka-X Could you generate the doc and put its screenshot to PR's description, please.

**Comment by Hisoka-X** [comment]
image
image

**Review by HyukjinKwon — APPROVED** [review]
Seems fine but I think it needs @cloud-fan 's approval.

File: docs/sql-ref-syntax-dml-insert-table.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ The `INSERT` statement inserts new rows into a table or overwrites the existing
 ### Syntax
 
 ```sql
-INSERT [ INTO | OVERWRITE ] [ TABLE ] table_identifier [ partition_spec ] [ ( column_list ) ]
-    { VALUES ( { value | NULL } [ , ... ] ) [ , ( ... ) ] | query }
+INSERT [ INTO | OVERWRITE ] [ TABLE ] table_identifier [ [ partition_spec ] [ ( column_list ) ]
+    { VALUES ( { value | NULL } [ , ... ] ) [ , ( ... ) ] | query } | REPLACE WHERE boolean_expression query ]
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-dml-insert-table.md:30]: This is not accurate. We should add a new SQL statement to document it
```
INSERT [ INTO | OVERWRITE ] [ TABLE ] table_identifier ...

INSERT INTO [ TABLE ] table_identifier REPLACE WHERE boolean_expression query ]
```

Comment (Hisoka-X) [reply docs/sql-ref-syntax-dml-insert-table.md]: Done

**Review by cloud-fan — APPROVED** [review]

**Pull request closed without merging**